calcom/cal.com
# feat: CRUD for oAuth users

## What does this PR do?

Allows creating users tied to a specific internal Cal oAuth, and then to fetch or update them. Since all managed users will be tied to an oAuth client, the routes are:
1. POST `/api/v2/oauth-clients/:clientId/users` - create new user tied to oAuth client
2. GET `/api/v2/oauth-clients/:clientId/users/:userId` - access oAuth client user
3. PUT `/api/v2/oauth-clients/:clientId/users/:userId` - update oAuth client user

The new routes can be found in the new [OAuthUserController](https://github.com/calcom/cal.com/pull/12853/files#diff-e9b3a49ec089b2b99e95994b4fd9b7dab220b9026ed020e42f6ffc658ec5ec2d).

Because of the decision to have `clientId` in the URL, the [OAuthClientGuard](apps/api/v2/src/modules/oauth/guard/oauth-client/oauth-client.guard.ts) was changed to access `clientId` from the URL and also it's tests [have been updated](https://github.com/calcom/cal.com/pull/12853/files#diff-73f42b57f202ac41d02143a48b59b30865e3c9d6f0819b41750fe6d1c5a75530).

Linear issue: [PRO-155](https://linear.app/calcom/issue/PRO-155/crud-platform-users-using-oauth-client)

## How should this be tested?

The new [OAuthUserController](https://github.com/calcom/cal.com/pull/12853/files#diff-e9b3a49ec089b2b99e95994b4fd9b7dab220b9026ed020e42f6ffc658ec5ec2d) has [e2e tests](https://github.com/calcom/cal.com/pull/12853/files#diff-aad793ae3f067850994d6a81afe4c9faf0c1dcbb2d29e0c368b1a4cac7e8defb). To test manually check test structure and make postman requests.
